### ISSUE TITLE
'var' keyword is highlighted differently when used in for loop.

### ISSUE THREAD

**Issue by safield** [Dart-Code/Dart-Code#3932]
In the following listed example, the var keyword will be highlighted with two different colors.

In the first case it is given semantic token 'keyword' and the other textmate scope 'storage.type.primitive.dart'.

I think it would make sense for the var keyword to be the same color always.

```
    var list = [];
    for (var item in list) {
      ...
    }
```

**Comment by DanTup** [comment]
Good catch, thanks! Our of interest, what theme are you using? They're the same in my theme (Dark+) so this wasn't visible.

Got a fix at https://dart-review.googlesource.com/c/sdk/+/241600/. It's in the LSP server so it'll show up in a Dart/Flutter SDK release (after merging) rather than an update to the VS Code extension. The `var` outside of the loop was correct, and the one inside the `for` was missing its semantic token.

(I'll close this issue once the change is merged into the SDK).

**Comment by safield** [comment]
I am using the ubuntu-monokai theme that can be found on my github page. It is based off the stock monokai, and most issues that symptomize on one, also symptomize on the other.

**Comment by DanTup** [comment]
Got it, thanks!

The fix for this has landed in https://github.com/dart-lang/sdk/commit/84fbee066e5bce430104812153579c20bfb1d027. It will appear in an upcoming SDK release.

I'll do some testing with the theme above, but if you notice any other inconsistencies before me/that I don't, please do file additional issues. Thanks!
